demo/lightning/checkpoint/multinode/strategies.py

```diff
@@ -19,28 +21,49 @@
                                                        GCSDistributedWriter)
 
 
-def save_checkpoint_helper(rank, checkpoint, path, checkpoint_io, writer):
+def checkpoint_helper(checkpoint):
+    """Flatten the model and optimizer states into a new dict.
+
+    Args:
+        checkpoint (dict): dict containing model and trainer state.
+
+    Returns:
+        converted_state (dict): Checkpoint state containing just the model and flattened optimizer states.
+
+        checkpoint (dict): Remaining metadata from the checkpoint.
+    """
     converted_state = {"model": checkpoint.pop("state_dict")}
     converted_state.update({
         f"optimizer_{idx}": optim_state
         for idx, optim_state in enumerate(
             checkpoint.pop("optimizer_states", []))
     })
-    save(converted_state, checkpoint_id=path, storage_writer=writer)
-
-    if rank == 0:
-        checkpoint_io.save_checkpoint(checkpoint, path / _METADATA_FILENAME)
+    return converted_state, checkpoint
 
 
 class DatafluxFSDPStrategy(FSDPStrategy):
 
-    def __init__(self, project_name, storage_client, **kwargs):
+    def __init__(self,
+                 project_name,
+                 storage_client,
+                 use_async=False,
+                 **kwargs):
         super().__init__(**kwargs)
         self.checkpoint_io = DatafluxLightningCheckpoint(
             project_name, storage_client)
         self.storage_client = storage.Client(project=project_name)
         user_agent.add_dataflux_user_agent(self.storage_client)
 
+        # Attributes used for async behavior.
+        self.use_async = use_async
+        self.checkpoint_group = None
+        self._checkpoint_future = None
+
+        if self.use_async:
+            default_ranks = list(range(dist.get_world_size()))
+            self.checkpoint_group = dist.new_group(
```

Comment:
Can you explain what does checkpoint_group do ? And why is it needed ? I dont see it being consumed here (but it may be used in some other layer)